lib/Runtime/Library/InJavascript/Array_prototype.js

```diff
@@ -27,15 +27,9 @@
     __chakraLibrary.ArrayIterator.prototype = CreateObject(iteratorPrototype);
     __chakraLibrary.raiseNeedObjectOfType = platform.raiseNeedObjectOfType;
     __chakraLibrary.raiseThis_NullOrUndefined = platform.raiseThis_NullOrUndefined;
-    __chakraLibrary.raiseNeedObject = platform.raiseNeedObject;
```

Comment:
Anything not needed for the Array methods was moved into separate files `Object_constructor.js` and `Math_object.js`.